packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -1043,6 +1047,44 @@ export class BasicCrawler<Context extends CrawlingContext = BasicCrawlingContext
         return kvs.getAutoSavedValue<State>(BasicCrawler.CRAWLEE_STATE_KEY, defaultValue);
     }
 
+    protected get pendingRequestCountApproximation(): number {
+        let result = 0;
+
+        if (this.requestQueue) {
+            result += this.requestQueue.getPendingCount();
+        }
+        if (this.requestList) {
+            result += this.requestList.length() - this.requestList.handledCount();
+        }
+
+        return result;
+    }
+
+    protected calculateEnqueuedRequestLimit(explicitLimit?: number): number | undefined {
```

Comment:
how about 
```suggestion
    protected calculateEnqueuedRequestLimit(explicitLimit: number = Infinity): number {
```

so we save some `requestLimit !== undefined` down the road?

Reply:
I actually had `Infinity` there and changed it because it was to mathematic-y. I can change it back :grin: